### PR TITLE
[nereids](array_agg)support array_agg in nereids

### PR DESCRIPTION
**Pull request by LemonLiTree**
support array_agg in nereids

### CODE REVIEW

**Comment by LemonLiTree** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/ArrayAgg.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    @Override
+    public List<FunctionSignature> getSignatures() {
+        return SIGNATURES;
+    }
+
+    @Override
+    public AggregateFunction withDistinctAndChildren(boolean distinct, List<Expression> children) {
+        Preconditions.checkArgument(children.size() == 1);
+        return new ArrayAgg(distinct, children.get(0));
+    }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/ArrayAgg.java:60]: need override accept function

Comment (LemonLiTree) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/ArrayAgg.java:56]: done

**Comment by LemonLiTree** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by keanji-x — APPROVED** [review]

**Pull request merged**